### PR TITLE
Add support for non-Ember Data based select API lookups

### PR DESCRIPTION
**Pull request by sandersky**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

* **Added** ability to drive select inputs via an API without using Ember Data. Instead you can do something like the following (this example lets you search a city and get a list of matching states from MapQuest's geolocation API):

  **Model**
  ```json
  {
    "properties": {
      "location": {
        "type": "string"
      },
      "results": {
        "type": "string"
      }
    },
    "type": "object"
  }
  ```

  *View*
  ```json
  {
    "properties": {
      "location": {
        "type": "string"
      },
      "results": {
        "endpoint": "http://www.mapquestapi.com/geocoding/v1/address",
        "labelAttribute": "adminArea3",
        "recordsPath": "results.0.locations",
        "query": {
          "key": "<mapquest-api-goes-here>",
          "location": "${./location}"
        },
        "type": "string",
        "valueAttribute": "linkId"
      }
    },
    "type": "object"
  }
  ```

### CODE REVIEW

File: addon/list-utils.js [hunk 1]
```diff
@@ -17,19 +18,38 @@ import _ from 'lodash'
...
-    return getAsyncDataValues(value, options, filteredData, bunsenId, store, filter)
+    return getItemsFromEmberData(value, options, filteredData, bunsenId, store, filter)
+  } else if (options.endpoint) {
+    return getItemsFromAjaxCall({ajax, data: filteredData, filter, options, value})
   }
 
   return RSVP.resolve(filteredData)
 }
 
+export function getQuery ({bunsenId, filter, query, value}) {
```
Comment (sophypal) [inline addon/list-utils.js]: jsdoc

File: addon/list-utils.js [hunk 2]
```diff
@@ -46,6 +66,41 @@ export function getEnumValues (values = [], filter = '') {
   return filteredValues
 }
 
+export function getItemsFromAjaxCall ({ajax, data, filter, options, value}) {
```
Comment (sophypal) [inline addon/list-utils.js]: jsdoc

File: addon/list-utils.js [hunk 3]
```diff
@@ -56,33 +111,38 @@ export function getEnumValues (values = [], filter = '') {
...
-      return data.concat(items)
+    .then((records) => {
+      return getItems({data, labelAttribute, records, valueAttribute})
     }).catch((err) => {
       Logger.log(`Error fetching ${modelType}`, err)
       throw err
     })
 }
+
+function getItems ({data, labelAttribute, records, valueAttribute}) {
```
Comment (sophypal) [inline addon/list-utils.js]: jsdoc

**Comment by sandersky** [comment]
:+1:

**Pull request merged**